### ISSUE TITLE
Connection Refused 

### ISSUE THREAD

**Issue by guru888** [acelaya/ZF-AcMailer#138]
Hi, I am new to Zend. I have connections for the mail server. Tested on the other project, working fine. I am trying to send the mail but it is throwing exception connection refused. Here is the exception can you please help me out. 

An error occurred. Exception: \n exception 'Zend\Mail\Protocol\Exception\RuntimeException' with message 'Connection refused' in /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Protocol/AbstractProtocol.php:209 Stack trace: #0 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Protocol/Smtp.php(148): Zend\Mail\Protocol\AbstractProtocol->_connect('tcp://127.0.0.1...') #1 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Transport/Smtp.php(375): Zend\Mail\Protocol\Smtp->connect() #2 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Transport/Smtp.php(361): Zend\Mail\Transport\Smtp->connect() #3 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Transport/Smtp.php(372): Zend\Mail\Transport\Smtp->lazyLoadConnection() #4 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mail/src/Transport/Smtp.php(229): Zend\Mail\Transport\Smtp->connect() #5 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/acelaya/zf2-acmailer/src/Service/MailService.php(95): Zend\Mail\Transport\Smtp->send(Object(Zend\Mail\Message)) #6 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/acelaya/zf2-acmailer/src/Controller/Plugin/SendMailPlugin.php(71): AcMailer\Service\MailService->send() #7 [internal function]: AcMailer\Controller\Plugin\SendMailPlugin->__invoke(Array) #8 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(309): call_user_func_array(Object(AcMailer\Controller\Plugin\SendMailPlugin), Array) #9 /Applications/XAMPP/xamppfiles/htdocs/monoland/module/Application/src/Application/Controller/EmailController.php(21): Zend\Mvc\Controller\AbstractController->__call('sendMail', Array) #10 /Applications/XAMPP/xamppfiles/htdocs/monoland/module/Application/src/Application/Controller/EmailController.php(21): Application\Controller\EmailController->sendMail(Array) #11 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(82): Application\Controller\EmailController->sendAction() #12 [internal function]: Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent)) #13 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func(Array, Object(Zend\Mvc\MvcEvent)) #14 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure)) #15 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(118): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure)) #16 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mvc/src/DispatchListener.php(93): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response)) #17 [internal function]: Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent)) #18 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func(Array, Object(Zend\Mvc\MvcEvent)) #19 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure)) #20 /Applications/XAMPP/xamppfiles/htdocs/monoland/vendor/zendframework/zend-mvc/src/Application.php(314): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure)) #21 /Applications/XAMPP/xamppfiles/htdocs/monoland/public/index.php(23): Zend\Mvc\Application->run() #22 {main}

Settings: for email

```php
<?php
return [
    'acmailer_options' => [

        // Default mail service
        'default' => [
            /*
             * Tells which other mail service configuration to extend.
             * The default service is usually the one that doesn't extend any other configuration, but others extend
             * from it.
             * By default, extends value is null.
             */
            'extends' => null,

            /***********
             * ADAPTER *
             ***********/

            /*
             * The mail adapter to be used.
             * You can define any class implementing Zend\Mail\Transport\TransportInterface,
             * either the fully qualified class name as string or the instance to be used.
             *
             * For standard mail transports, you can use these aliases:
             *      - sendmail  => Zend\Mail\Transport\Sendmail
             *      - smtp      => Zend\Mail\Transport\Smtp
             *      - file      => Zend\Mail\Transport\File
             *      - null      => Zend\Mail\Transport\Null (<=ZF2.3) or Zend\Mail\Transport\InMemory (>=ZF2.4)
             *      - in_memory => Zend\Mail\Transport\Null (<=ZF2.3) or Zend\Mail\Transport\InMemory (>=ZF2.4)
             *
             * If for some reason you need to use a custom adapter which is complex to be created, you can also define
             * the name of a service that returns a Zend\Mail\Transport\TransportInterface instance. When the
             * MailService is created, it will automatically check if this is an existing service.
             *
             * Default value is \Zend\Mail\Transport\Sendmail
             */
            //'mail_adapter' => '\Zend\Mail\Transport\Sendmail',
            'mail_adapter' => 'Zend\Mail\Transport\Smtp',

            /*
             * An alias for the 'mail_adapter' option. Use just one or another.
             */
            //'transport' => '\Zend\Mail\Transport\Sendmail',


            /************************
             * MESSAGE CONFIGURATION *
             ************************/

            'message_options' => [
                /*
                 * From email address of the email.
                 *
                 * Default value is an empty string
                 */
                //'from' => '',
                'from' => 'no-reply@monoland.com',

                /*
                 * From name to be displayed instead of from address.
                 *
                 * Default value is an empty string
                 */
                //'from_name' => '',
                //'from_name' => 'Monoland System Mailer',

                /*
                 * Destination addresses of sent emails.
                 * It can be an email address as string or an array of email addresses.
                 *
                 * Default value is an empty array.
                 */
                'to' => ['guru@newtonsit.com.au'],

                /*
                 * Copy destination addresses of sent emails.
                 * It can be an email address as string or an array of email addresses.
                 *
                 * Default value is an empty array
                 */
                //'cc' => [],

                /*
                 * Hidden copy destination addresses of sent emails.
                 * It can be an email address as string or an array of email addresses.
                 *
                 * Default value is an empty array
                 */
                //'bcc' => [],

                /*
                 * Email subject.
                 *
                 * Default value is an empty string
                 */
                //'subject' => '',

                /*
                 * Email body.
                 * It can be a plain-text string, a HTML string or a template that will be computed at runtime.
                 * The 'content' property is used by default.
                 * To force a template configuration to be used, set 'use_template' to true.
                 *
                 * Default content is an empty string.
                 */
                //'body' => [
                    //'content' => '',
                    //'charset' => 'utf-8',
                    //'use_template' => false,

                    /*
                     * Defines information to create the email body from a view partial.
                     * It defines template path and template params.
                     * The path will be resolved by a view resolver, so you need to place mail templates inside a view
                     * folder of one of your modules or customize your template map and template path stack.
                     * Params will be a group of key-value pairs.
                     *
                     * The 'children' property allows to define children for the template, in case you want to use
                     * layouts.
                     * You can define any number of children. The key in the array will be used as the 'capture_to'
                     * property when rendering the template.
                     * If you set the key 'content' to the child, you should have something like echo $this->content in
                     * you layout.
                     *
                     * Any child can have its own children, so you can nest views into other views recursively.
                     *
                     * By default no children are used
                     */
                    //'template' => [
                    //    'path'          => 'ac-mailer/mail-templates/layout',
                    //    'params'        => [],
                    //    'children'      => [
                    //        'content'   => [
                    //            'path'   => 'ac-mailer/mail-templates/mail',
                    //            'params' => [],
                    //        ]
                    //    ],
                    //    'default_layout' => [
                    //        'path' => null,
                    //        'params' => [],
                    //        'template_capture_to' => 'content'
                    //    ]
                    //],
                //],

                /*
                 * Attachments config.
                 * Allows to define an array of files that will be attached to the message,
                 * or even a directory that will be iterated to attach all found files.
                 * Set directory will only be iterated if 'iterate' property is true and 'path' is a valid
                 * directory.
                 * If 'recursive' is true all nested directories will be iterated too.
                 * If both files and dir are set, all files will be merged without duplication.
                 *
                 * By default the files array is empty and the directory won't be iterated
                 */
                //'attachments' => [
                //    'files' => [],
                //    'dir' => [
                //        'iterate'   => false,
                //        'path'      => 'data/mail/attachments',
                //        'recursive' => false,
                //    ],
                //],
            ],


            /**********************
             * SMTP CONFIGURATION *
             **********************/

            /*
             * SMTP concrete options that will be used only when the adapter is a Zend\Mail\Transport\Smtp.
             * This will be ignored otherwise.
             */
            'smtp_options' => [
                /*
                 * Hostname or IP address of the mail server.
                 *
                 * Default value is localhost
                 */
                'host' => 'smtp.gmail.com',

                /*
                 * Port of the mail server.
                 *
                 * Default value is 25
                 */
                'port' => 587,

                /*
                 * The connection class used for authentication.
                 * The value can be one of 'smtp', 'plain', 'login' or 'crammd5'.
                 *
                 * Default value is 'smtp'.
                 */
                //'connection_class' => 'smtp',
                'connection_class' => 'login',

                'connection_config' => [
                    /*
                     * The SMTP authentication identity.
                     * If this is not set, the 'from' option of the message will be used.
                     *
                     * Default value is an empty string
                     */
                    //'username' => '',
                    'username' => 'wdwpostmaster@gmail.com',

                    /*
                     * The SMTP authentication credential.
                     *
                     * Default value is an empty string
                     */
                    //'password' => '',
                    'password' => 'asdasd',

                    /*
                     * This defines the encryption type to be used, 'ssl' or 'tls'.
                     * Boolean false should be used to disable SSL.
                     *
                     * Default value is false
                     */
                    'ssl' => 'tls',
                ],
            ],

   
            ],

        ]


    ]
];
```

**Comment by acelaya** [comment]
There's an extra closing bracket at the end of your configuration, but I suppose that's just a typo, because it would produce another kind of error.

Besides that, could you preformat the exception? I's hard to follow it this way.

**Comment by acelaya** [comment]
I'm closing this due to lack of activity
